firefoxThemes/popup/popup.js

```diff
@@ -165,66 +193,43 @@ function getMixState(name, mixTabs) {
 
 /*Initializes checkbox switches*/
 function prepareSwitches(storage) {
-  backgroundSwitch.checked = !!storage.backgroundType;
-  backgroundSwitch.disabled = !(!!storage.hasSecondaryMode);
-  if (storage.darkMode !== undefined) {
-    darkModeSwitch.checked = !!storage.darkMode.isDarkNow
-    darkModeSwitch.disabled = !!storage.darkMode.shouldDisable;
+  if(storage.systemTheme === systemStates.DRUTHERS){
+      selectTag.disabled = true;
+      darkModeSwitch.disabled = true;
+      backgroundSwitch.disabled = true;
   } else {
-    darkModeSwitch.checked = false;
-    darkModeSwitch.disabled = true;
+    backgroundSwitch.checked = !!storage.backgroundType;
+    backgroundSwitch.disabled = !(!!storage.hasSecondaryMode);
+    if (storage.darkMode !== undefined) {
+      darkModeSwitch.checked = !!storage.darkMode.isDarkNow
+      darkModeSwitch.disabled = !!storage.darkMode.shouldDisable;
+    } else {
+      darkModeSwitch.checked = false;
+      darkModeSwitch.disabled = true;
+    }
+    showSearchSwitch.checked = storage.showWidget === undefined || storage.showWidget;
```

Comment:
Something goofy happened when the theme gets applied from the settings then visiting the popup.

https://user-images.githubusercontent.com/15972415/146850769-85ebdef2-c298-4890-9442-286478a01da6.mp4

Reply:
For the *goofy part* do you mean the widget switch not initially syncing with the actual widget? If so, this ~feature~ bug has been there for sometime now. It only happens when you **first** use the extension, after that its smooth sailing.

Reply:
**First**, as in toggle the widget switch for the first time, of course.

Reply:
@Unthrottled  As a reminder, you can also click on the Doki Heart in the *Popup menu* to get to the *Options Page* as well.

Reply:
Huh, I remember that you did that along time ago, I totally forgot about the Heart link. Looks like `openOptionsPage`  keeps the option page open when the theme changes, neat!

Guess I never thought to click it, because I guess I just thought hovering make the moving heart grow bigger. Might have been a bit distracted 😄 

Maybe it may help to have like a hyperlink somewhere that says `View Settings` or something later?